snakemake/io.py

```diff
@@ -923,11 +939,12 @@ def flatten(wildcards):
                 values = [values]
             yield [(wildcard, value) for value in values]
 
+    formatter = string.Formatter()
     try:
         return [
-            filepattern.format(**comb)
```

Comment:
with `from functools import partial` this would become:
```
partial(filepattern.format, **comb)
```

Reply:
I don't quite understand how this works.
First, the line you suggested the change on is the original line, the new code is
``formatter.vformat(filepattern, (), comb)``
which allows partial formatting.
It might be easier to just always allow missing but I don't see how partial function helps here.

Reply:
In the end every target will have to have all wildcards completed, in this sense `allow_missing` is a bit misleading, since it's needed only to carry over partially-completed strings across rules/functions. The `functools.partial` would allow to complete them at different points in the code. However, I can see now that implementing this with partials might require a more substantial change in the code (perhaps something to look into in the future). Otherwise, yes, this is a great feature to have and the `allow_missing` solution seems very workable.

Reply:
I agree. It's a very useful feature.